src/frontend/src/lib/utils/token-card.utils.ts

```diff
@@ -23,3 +24,8 @@ export const mapHeaderData = ({
 	usdBalance,
 	tokenCount: tokens.length
 });
+
+export const isCardDataTogglableToken = (data: CardData): boolean => {
```

Comment:
nitpick: if you do

```suggestion
export const isCardDataTogglableToken = (data: CardData): data is TokenToggleable<Token> => {
```

you don't need to cast afterwards:

```ts
let token: TokenToggleable<Token> | undefined = $derived(
		isCardDataTogglableToken(data) ? data : undefined
	);
```